src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepDynamicContext.java

```diff
@@ -97,8 +97,6 @@ void resume(StepContext context) throws Exception {
         if (item == null) {
             throw new IllegalStateException("queue refused " + task);
         }
-        // Try to avoid having distinct a instance of PlaceholderTask here compared to any previously-scheduled task.
-        task = (ExecutorStepExecution.PlaceholderTask)item.task;
```

Comment:
Reverting this bit of #344 which should now be unnecessary: any instances in memory ought to be interchangeable at this point.